### ISSUE TITLE
Extend FileLoader with custom key

### ISSUE THREAD

**Issue by tobiasKaminsky** [bumptech/glide#3198]
Our [app](https://github.com/nextcloud/android) stores files in the cloud.
To save bandwith, it is possible to only download a screen adapted image of the real file.
Now I am changing our old caching system to glide.
The first part is working fine and I use our file eTag as identifier:
```
public LoadData<InputStream> buildLoadData(@NonNull GlideContainer container, int width, int height,
                                               @NonNull Options options) {
        return new LoadData<>(container.eTag, new HttpStreamGlideContainerFetcher(container));
```

But you can also download the image directly and view it (then no screen adapted image is generated).
If you then delete the local copy of the file, I want to create & store a screen adapted image from the local file.
My problem is now, that it must have exactly the same key as if it would have been downloaded.
As I read the signature is in addition, so it would not be the same.

So I tried to extend FileLoader, pass the eTag to buildLoadData, but the problem is that FileFetcher is private:
```
public class OCFileModelLoader extends FileLoader<OCFile> {
…
@Override
    public LoadData<OCFile> buildLoadData(@NonNull File model, int width, int height, @NonNull Options options) {
        return new LoadData<>(eTag, new FileFetcher<>(model, fileOpener));
    }
…
```
Of course I can copy FileFetcher as it is not that big class, but maybe there is a cleaner/better way?

tl;dr: I want to cache an image from a file with a specific key (no signature), but FileFetcher is private.

**Comment by tobiasKaminsky** [comment]
Just for information: I copied and adapted FileFetcher.
